### PR TITLE
Support configurable resource requests/limits for components

### PR DESCRIPTION
**Pull request by dseapy**
this approach is minimizes changes to existing values.yaml for backward compatibility.  however there are a few changes that seem preferable if we don't mind refactoring it a bit:

1. remove `controller` from the `component` section and simply have `image:` section at root level (which would be consistent with the rest of the `controller` config (i.e. resources, replicaCount, tolerations, affinity, etc)
2. remove `pullPolicy` from component `image` section, or support passing it through for components, as it currently doesn't appear to be used (other than for controller).
3. refactor to be of the following structure or similar (having the component name under the components key seems more natural & more concise as number of settings increases - ie. adding things like `priorityClassName` in the future would result in only 1 line per component being added):
```yaml
components:
  tsdb:
    image: ...
    resources: ...
  applier:
...
```

### CODE REVIEW

**Comment by bradbeam** [comment]
would you mind splitting out the args fix separately? I think we can get that through pretty quick to clean up the break, but haven't had a digest the rest of the changes yet

**Comment by dseapy** [comment]
> would you mind splitting out the args fix separately? I think we can get that through pretty quick to clean up the break, but haven't had a digest the rest of the changes yet

done (https://github.com/thestormforge/helm-charts/pull/28).  assuming it's ok i threw a bit into gitignore there too.

File: charts/optimize-live/templates/deployment.yaml [hunk 1]
```diff
@@ -34,33 +35,28 @@ spec:
...
           - --zap-log-level=9
-{{- end }}
+          {{- end }}
           env:
           - name: IMAGE_PULL_SECRET_NAME
             value: {{ include "optimize-live.fullname" . }}-docker
           - name: SF_CREDS_SECRET_NAME
             value: {{ include "optimize-live.fullname" . }}
 {{- if .Values.metricsURL }}
-{{- if (ne "" .Values.metricsURL) }}
```
Comment (dseapy) [inline charts/optimize-live/templates/deployment.yaml:47]: these are unrelated to PR, can break out to separate PR if preferred.  just done to make it a bit simpler to read.

File: charts/optimize-live/templates/deployment.yaml [hunk 2]
```diff
@@ -99,6 +95,14 @@ spec:
             {{- toYaml .Values.readinessProbe | nindent 12 }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+          - name: config
+            mountPath: "/config"
```
Comment (dseapy) [inline charts/optimize-live/templates/deployment.yaml:100]: @bradbeam mentioned use subpath and rm /config, ref: https://github.com/gramLabs/optimize-live-controller/blob/main/pkg/components/tsdb/deployment.go#L93-L97

File: charts/optimize-live/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.3.0
+version: 0.3.1
```
Comment (dseapy) [inline charts/optimize-live/Chart.yaml:18]: is comment accurate for our versioning strategy?

if semver & bumping each time, i'm guessing this should be 0.4.0.

Comment (dseapy) [reply charts/optimize-live/Chart.yaml:18]: also i'm guessing controller image tag would need to be updated along with this PR

**Review by bradbeam — APPROVED** [review]
